### PR TITLE
Implement Create Space As Profile

### PR DESCRIPTION
**Pull request by teodorus-nathaniel**
# Changes
- Remove opening switch space as profile modal after onboarding
- Make a variation of edit/create space with query param `as-profile=true`, which indicates edit/create profile instead of usual space. Changes the text from edit/create space to edit/create profile and extrinsic for create new profile
- Update links that goes to edit/create profile by adding the query param `as-profile`
- Reload profile after edit/create profile
- Update extrinsic used in create profile onboarding
- Change `+ create space` button in navbar to `+ create profile`
- Change `+ create profile space` button in profile (if no profile yet) to `+ create profile`
- Add soonsocial graphql url to staging connection config
- Open onboarding energy step if user wants to do tx but no energy

# Note
There are one `as any` casting because subsocial sdk in current version haven't had typing for createSpaceAsProfile extrinsic.
(fixed: updated subsocial packages to newest version)

# Merge Note
Need to remove commits prefixed with DELETE

### CODE REVIEW

File: src/components/profiles/address-views/utils/index.tsx [hunk 1]
```diff
@@ -33,14 +33,18 @@ type ProfileSpaceProps = BareProps & {
...
+      href={'/[spaceId]/edit' + asProfileQuery}
+      as={editSpaceUrl(profileSpace.struct) + asProfileQuery}
+    >
       <a className='item' onClick={onClick} {...props}>
         Edit profile
       </a>
     </Link>
   ) : (
-    <Link href='/spaces/new' as='/spaces/new'>
+    <Link href={'/spaces/new' + asProfileQuery} as={'/spaces/new' + asProfileQuery}>
```
Comment (olehmell) [inline src/components/profiles/address-views/utils/index.tsx:47]: Maybe better just create const like `newSpaceAsProfile` and put to both places?

File: src/components/spaces/helpers/CreateSpaceButton.tsx [hunk 2]
```diff
@@ -3,22 +3,25 @@ import { ButtonProps } from 'antd/lib/button'
 import { ButtonLink, IconLink } from 'src/components/utils/CustomLinks'
 import { BareProps } from 'src/components/utils/types'
 
-const newSpacePath = '/spaces/new'
-const title = 'Create space'
+const getNewSpacePath = (asProfile?: boolean) =>
+  '/spaces/new' + (asProfile ? '?as-profile=true' : '')
+const getTitle = (asProfile?: boolean) => `Create ${asProfile ? 'profile' : 'space'}`
```
Comment (olehmell) [inline src/components/spaces/helpers/CreateSpaceButton.tsx:8]: You use the code `asProfile ? 'profile' : 'space'` in lot of places, so it is not good for long support, better if you create small function for it

File: src/rtk/features/onBoarding/onBoardingHooks.ts [hunk 3]
```diff
@@ -106,9 +105,9 @@ const getSavedProfileData = (profile: SpaceContent | undefined) => {
   return { about, email, image, links, name, tags }
 }
 const upsertProfile = (api: ApiPromise, cid: IpfsCid, currentProfileId?: string) => {
-  const permissions = newWritePermission()
   if (!currentProfileId) {
-    return api.tx.spaces.createSpace(IpfsContent(cid?.toString()), permissions)
+    // TODO: remove any cast when sdk is updated
```
Comment (olehmell) [inline src/rtk/features/onBoarding/onBoardingHooks.ts:109]: Create task and link with it
https://app.clickup.com/t/8677d8xh5

**Comment by teodorus-nathaniel** [comment]
@olehmell I thought it will be better to remove as props from the links, because after I check in next docs (both [new one](https://nextjs.org/docs/api-reference/next/link) and [the version we use](https://nextjs.org/docs/tag/v11.1.4/api-reference/next/link#dynamic-routes)), it doesn't explain as props and I don't think its needed for our case.

File: src/components/spaces/helpers/CreateSpaceButton.tsx (hunk 2 above)
Comment (teodorus-nathaniel) [reply src/components/spaces/helpers/CreateSpaceButton.tsx:8]: After I edit the url links, and refactor the one in edit space, now this type of code is only in two places. I think it's not necessary to create a wrapper for the wording in this case

File: src/rtk/features/onBoarding/onBoardingHooks.ts (hunk 3 above)
Comment (teodorus-nathaniel) [reply src/rtk/features/onBoarding/onBoardingHooks.ts:109]: Created the PR in https://github.com/dappforce/subsocial-js/pull/127

**Review by olehmell — APPROVED** [review]

**Pull request merged**